example-todomvc/src/main/scala/mhtml/todo/MhtmlTodo.scala

```diff
@@ -0,0 +1,342 @@
+  val LocalStorageName = "todo.mhtml"
+  def load(): Seq[Todo] =
+    LocalStorage(LocalStorageName).toSeq.flatMap(read[Seq[Todo]])
+  def save(todos: Seq[Todo]): Unit =
+    LocalStorage(LocalStorageName) = write(todos)
+  val windowHash: Rx[String] = Rx(dom.window.location.hash).merge{
+    var updatedHash = Var(dom.window.location.hash)
+    dom.window.onhashchange = (ev: HashChangeEvent) => {
+      updatedHash := dom.window.location.hash
+    }
+    updatedHash
+  }
+  //lazy val autoSave: Unit = allTodos.foreach(save)
+  val editingTodo: Var[Option[Todo]] = Var[Option[Todo]](None)
+  val all = TodoList("All", "#/", allTodos)
+  val active =
+    TodoList("Active", "#/active", allTodos.map(_.filter(!_.completed)))
+  val completed =
+    TodoList("Completed", "#/completed", allTodos.map(_.filter(_.completed)))
+  val todoLists = Seq(all, active, completed)
+
+
+  val currentTodoList: Rx[TodoList] = windowHash.map(hash =>
+    todoLists.find(_.hash === hash).getOrElse(all)
+  )
+  val todoListComponents: Rx[Seq[Component[TodoListItemData]]] =
+    currentTodoList.flatMap { current =>
+      current.items.map(_.map(todoListItem))
+    }
+
+  // Note: Cats Traverse can't support Seq, so we use List
+  def unzipListComponents(listComps: Seq[Component[TodoListItemData]])
+  : (List[Node], List[Rx[TodoListItemData]])
+  = listComps.toList.map(tlc => (tlc.view, tlc.model)).unzip
+
+  val todoListElems: Rx[List[Node]] =
+    todoListComponents.map{tlcSeq => unzipListComponents(tlcSeq)._1}
+
+  val todoListDataChanges: Rx[List[TodoListItemData]] =
+    todoListComponents.flatMap{tlcSeq => unzipListComponents(tlcSeq)._2.sequence}
+
+//  lazy val allTodos: Rx[Seq[Todo]] = (
+//    Rx(load()) |@| header.model |@| todoListDataChanges
+//  ) map {
+//    case (stored: Seq[Todo], newTodoMaybe: Option[Todo], changes: List[TodoListItemData]) =>
+//      val removals: List[Todo] = changes.flatMap(change => change.removal)
+//      val updates: List[TodoUpdate] = changes.flatMap(change => change.update)
+//
+//      (stored ++ newTodoMaybe).filter(todo => !removals .contains(todo))
+//      .map{todo => updates.find{update => update.oldTodo == todo} match {
+//        case Some(foundUpdate) => foundUpdate.newTodo
+//        case None => todo
+//      }}
+//  }
+
+  case class ModelSources(
+    headerModel: Option[Todo],
+    changes:  List[TodoListItemData]
+  )
+
+  def updateState(currentTodos: Seq[Todo], sources: ModelSources): Seq[Todo] = {
+    val removals: List[Todo] = sources.changes.flatMap(change => change.removal)
+    val updates: List[TodoUpdate] = sources.changes.flatMap(change => change.update)
+
+    (currentTodos ++ sources.headerModel).filter(todo => !removals .contains(todo))
+      .map{todo => updates.find{update => update.oldTodo == todo} match {
+        case Some(foundUpdate) => foundUpdate.newTodo
+        case None => todo
+      }}
+  }
+
+  val sourcesWrapped: Rx[ModelSources] = header.model |@| todoListDataChanges map {
```

Comment:
I was wondering about that as well since it seemed that everything could be handled as single updates - thanks for the suggestion! Will try next time.